apps/aecore/src/aec_db.erl

```diff
@@ -719,28 +756,39 @@ add_index_plugins() ->
     ok.
 
 ensure_mnesia_tables(Mode, Storage) ->
-    Tables = tables(Mode),
+    TablesMandatory = tables_mandatory(Mode),
+    TablesOptional = tables_created_if_not_present(Mode),
     case Storage of
         existing_schema ->
-            case check_mnesia_tables(Tables, []) of
-                [] -> ok;
+            case check_mnesia_tables(TablesMandatory, [], []) of
+                [] ->
+                    case check_mnesia_tables(TablesOptional, [], [{create_missing_tables, true}]) of
+                        [] -> ok;
+                        Errors ->
+                            lager:error("Database check failed: ~p", [Errors]),
```

Comment:
Is lager guaranteed to be started at this stage? Problem may be present in existing code too.

Reply:
It is not, and - yes - this is an issue in the existing code as well. I just coded this this way to keep it consistent.